uaclient/contract.py

```diff
@@ -309,9 +309,29 @@ def process_entitlements_delta(
     :param series_overrides: Boolean set True if series overrides should be
         applied to the new_access dict.
     """
+    from uaclient.entitlements import entitlements_enable_order
+
     delta_error = False
     unexpected_error = False
-    for name, new_entitlement in sorted(new_entitlements.items()):
+
+    # We need to sort our entitlements because some of them
+    # depend on other service to be enable first. Additionally,
+    # we must be careful because no all entitlements that are
+    # provided by the contract are supported by UA.
+    def sort_entitlements(key_name):
+        ent_order = entitlements_enable_order(cfg)
```

Comment:
If we don't do the refactor I suggest below/above, then we should only call `entitlements_enable_order` once by moving it outside of the `sort_entitlements` function

Reply:
Actually, I guess this isn't a big deal since its only recursing over a list <10 items long. Same for below similar suggestion